library/src/main/java/com/okta/oidc/clients/sessions/SessionClientImpl.java

```diff
@@ -126,6 +127,25 @@ public void authorizedRequest(@NonNull Uri uri, @Nullable Map<String, String> pr
         });
     }
 
+    @Override
+    public void authorizedRequest(@NonNull Uri uri, @Nullable Map<String, String> properties,
+                                  @Nullable Map<String, String> postParameters,
+                                  @Nullable byte[] requestBody,
+                                  @NonNull ConnectionParameters.RequestMethod method,
+                                  final RequestCallback<ByteBuffer, AuthorizationException> cb) {
+        cancelFuture();
```

Comment:
This is huge limitation. One http request at a time???

Reply:
Yes since the default implementation provide one networking thread for the async client. Users can use the SyncSessionClient and implement their own threading mechanism if they want to do multiple requests at the same time.

Reply:
Then it doesn't make any sense, use `SyncSessionClient` and implement `own threading mechanism`,  it is too complicated. There are no applications today that don't send http requests in parallel. As a developer I would prefer just to use http library that do all multi-threading and managing of multiple requests for me.
Current implementation is error prone

Reply:
Yes I understand what you're saying, users should write their own connection code to do this but it is one of the requirements in OKTA-240108. This API is a helper for that append the bearer token to http requests. It's for users that don't want to use a separate library or write own HttpURLConnection. Not intended to be a http library. This API doesn't prevent the application from doing parallel http requests. It prevents it from doing parallel authorizedRequest calls only if they use the async client.

Reply:
My own expectation is if I use AsyncClient for sending `authorized` http requests with arbitrary payload, it should satisfy the following requirements:
1. It should asynchronously return data or error
2. It should handle multiple requests

If requirement 2 can't be implemented, then probably it is better to remove this implementation from AsyncClient, as again, this implementation can cause unpredictable behaviors and source of the issue can be really hard to found

Reply:
2. Can be implemented but it's beyond the scope of this PR and ticket. Since we'll have to take care of queueing, thread priorities, clean up. I can remove the cancel and it will run sequentially.

Reply:
I think that the serial queue of http requests is better solution than cancellation. We probably should add a **note** in readme file about implementation details